submissions/7SergSmith7/Tiny JS World/index.js

```diff
@@ -0,0 +1,125 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Habitant {
+  constructor(species, name, gender, phrase, friends = [], legs, hands) {
+    this.species = species;
+    this.legs = legs;
+    this.hands = hands;
+    this.name = name;
+    this.gender = gender;
+    this.phrase = phrase;
+    this.friends = friends;
+    //this.addFriends(friends);
+  }
+
+  addFriends(newFriend) {
+    this.friends.push(newFriend);
+  }
+
+  listFriends() {
+    return this.friends.length ? this.friends.join(", ") : "I am loner";
+  }
+
+  sayPhrase() {
+    return this.phrase;
+  }
+
+  toString() {
+    return `Hi! I am <strong>${this.species}</strong>, my name is <strong>${
+      this.name
+    }</strong>, my gender is <strong>${this.gender}</strong>, I have <strong>${
+      this.legs
+    }</strong> legs and <strong>${this.hands}</strong> hands. 
+     My friends: <strong>${this.listFriends()}</strong> and my phrase: <strong>${this.sayPhrase()}</strong>`;
+  }
+}
+
+class Dog extends Habitant {
+  constructor(name, gender, phrase, friends, legs = 4, hands = 0) {
+    super("dog", name, gender, phrase, friends, legs, hands);
+  }
+  toString() {
+    return super.toString();
+  }
+  sayPhrase() {
+    return super.sayPhrase();
+  }
+  addFriends(newFriend) {
+    super.addFriends(newFriend);
+  }
+  listFriends() {
+    return super.listFriends();
+  }
```

Comment:
You don't need to re-define any methods defined in a super class, they are inherited and automatically available to the child classes however deep hierarchy is.

Yet you can still redefine `toString` to add value unique to a child class, in this task.

Reply:
done

Reply:
👍  Code became shorter and concise )